cmake/EnableSanitizers.cmake

```diff
@@ -38,44 +38,64 @@ option(SANITIZE_ADDRESS "Enable AddressSanitizer for the build." "")
 if (SANITIZE_ADDRESS)
     set(ASAN_CANDIDATES "-fsanitize=address -fno-omit-frame-pointer" "-fsanitize=address")
     sanitizer_test(AddressSanitizer ${ASAN_CANDIDATES})
+    if (NOT AddressSanitizer_ENABLED)
+        message(FATAL_ERROR "AddressSanitizer requested but could not be enabled.")
```

Comment:
How can this happen? Due to a user error, configuration error, or incompatibility between options?

This message is informative but not actionable. What do you want the user to do after they see this? Can they fix it somehow? Do you want them to file a bug?

Reply:
I added some more explanations.